plugins/services/src/js/containers/services/ServicesContainer.js

```diff
@@ -432,28 +432,54 @@ class ServicesContainer extends React.Component {
     };
   }
 
-  getModals(service) {
-    const modalProps = Object.assign({}, this.state.modal);
+  /**
+   * This function cleans and updates the modalProps property
+   *
+   * This is needed to refresh the state of the service from the store once the
+   * modal is loaded. In our delete group modal for example we need feedback
+   * of the service while it is being deleted. In other places (after deleting
+   * an app) on the other hand, it is possible that the modal is still open,
+   * DCOSStore returns undefined but we still need the (old) information stored
+   * in modalProps.service - and want to close the modal.
+   *
+   * @param {object} stateService - an object which contains id and a service
+   * @param {ServiceTree} [service] - service information
+   * @returns {object} updated and cleaned up modal information (props)
+   */
+  getUpdatedModalProps(stateService, service) {
```

Comment:
Maybe rename this to be more specific for what the method is doing?

Reply:
hm, any Idea how this should be named instead?

Reply:
Something like `getModalWithPropsWithCorrectedService`.